community_supplied/VendorSpecific/Paloalto/check-paloalto-processes-cpu-memory-netconf.rule

```diff
@@ -0,0 +1,272 @@
+/*
+ * Detects PANOS processes (daemons) CPU and memory utilization threshold breaches and
+ * notifies when anomalies are found.
+ * Three inputs control detection
+ * 
+ *   1) process-name-input, is a regular expression that matches the daemons that you
+ *      would like to monitor. By default it's '.*', which matches all daemons.
+ *      Use something like 'rpdbind|syslogd|authd' to match only rotuing, firewall and
+ *      interface daemons.
+ * 
+ *   2) process-high-threshold, is the threshold that causes the rule to 
+ *      report an anomaly. By default it's 80% of daemon CPU and Memory utilization. This
+ *      rule will set a dashboard color to red when daemon CPU and Memory utilization
+ *      exceed high threshold 'process-high-threshold' for period of
+ *      3 minutes.
+ * 
+ *   3) process-low-threshold, is the threshold that causes the rule to 
+ *      report an anomaly. By default it's 50% of daemon CPU and Memory utilization. This
+ *      rule will set a dashboard color to yellow when daemon CPU and Memory utilization
+ *      exceed low threshold 'process-low-threshold' for period of
+ *      3 minutes. Otherwise color is set to green.
+ */
+healthbot {
+    topic system.processes {
+        description "Monitors system statistics i.e. CPU, memory, storage, processes & queues and notifies anomalies";
+        synopsis "System statistics analyzer";
+        rule check-paloalto-processes-cpu-memory-netconf {
+            description "Collects system process's cpu utilization periodically and notifies anomalies when daemon CPU usage exceed threshold";
+            synopsis "System processes cpu analyzer";
+            /*
+             * Monitors *all* the daemons CPU utilization. Notifies via the
+             * dashboard colors when the daemon CPU utilization exceed by
+             * {{process-cpu-high-threshold}} for every collected metric during
+             * a 3 minute time range (red). The color is set to yellow if we
+             * see daemon CPU utilization exceed {{process-cpu-low-threshold}}
+             * for every collected metric during a 3 min time range. Otherwise 
+             * the color is set to green.
+             * 
+             * Use daemon name and pid as key for rule.
+             */
+            keys [ daemon pid ];
+            /*
+             * Sensor configuration to get data from network devices.
+             */
+            sensor systemprocexttable {
+                synopsis "System processes CPU iAgent sensor definition";
+                description "iAgent netconf rpc/cli command sensor to collect data from network device";
+                iAgent {
+                    file PaloaltoSysytemResource.yml;
+                    table PaloaltoSysytemResourceTable;
+                    frequency 60s;
```

Comment:
And this..

Reply:
Offset is only for trigger frequency and time-range and not for sensor in rule. For sensor, we need to configure under ingest settings.